### PR TITLE
[GLUTEN-3258][DOC] Correct a malformed table for two spark functions

### PR DESCRIPTION
**Pull request by zwangsheng**
## What changes were proposed in this pull request?

Corrent bitwise_or function symbol and malposition in `./docs/velox-backend-support-progress.md`

Use `&#124; ` as an escape from `|`.

## How was this patch tested?

In local markdown preview.

### CODE REVIEW

File: docs/velox-backend-support-progress.md [hunk 1]
```diff
@@ -104,8 +104,7 @@ Gluten supports 164 functions. (Draw to right to see all data types)
 | >                             | gt                     | greaterthan           | S      |                        | S       | S    | S     | S   | S    | S     | S      |      |           | S      |         |      |        |          |       |      |        |      |
 | >=                            | gte                    | greaterthanorequal    | S      |                        | S       | S    | S     | S   | S    | S     | S      |      |           | S      |         |      |        |          |       |      |        |      |
 | ^                             | bitwise_xor            |                       | S      |                        |         |      | S     | S   | S    |       |        |      |           |        |         |      |        |          |       |      |        |      |
-| \                             |                        | bitwise_or            | bitwise_or            | S      |                        |         |      | S     | S   | S    |       |        |      |           |        |         |      |        |          |       |      |        |      |
```
Comment (PHILO-HE) [inline docs/velox-backend-support-progress.md:107]: Yes, the form was incorrectly changed at https://github.com/oap-project/gluten/commit/adb1bf05038cb60c10756936280bc624ee3d3c0e.
It looks we have to keep `\` as escape char to present `|` symbol for this function. Could you refine this patch? Thanks!

File: docs/velox-backend-support-progress.md [hunk 2]
```diff
@@ -104,8 +104,7 @@ Gluten supports 164 functions. (Draw to right to see all data types)
...
-| \                             | \                      |                       |                        |                       |        |                        |         |      |       |     |      |       |        |      |           |        |         |      |        |          |       |      |        |      |
```
Comment (PHILO-HE) [inline docs/velox-backend-support-progress.md:108]: This line may need to be kept for `||` function with some changes. See [link](https://spark.apache.org/docs/latest/api/sql/index.html#_18).

Comment (zwangsheng) [reply docs/velox-backend-support-progress.md:108]: Thanks for your info, i will modify this based on the above info.

**Comment by zwangsheng** [comment]
Hi @PHILO-HE, addressed both `|` and `||`, in local preview like:

截屏2023-09-25 10 12 26

**Review by PHILO-HE — APPROVED** [review]
This fix is better. Thanks!

**Pull request merged**